scripts/update-ios-examples.mjs

```diff
@@ -0,0 +1,136 @@
+/**
+ * This script updates the iOS examples part of the DocC documentation.
+ * See platform/ios/CONTRIBUTING.md for the details.
+ */
+
+import fs from "fs/promises";
+
+/** documentation files that will be updated */
+const documentationPaths = [
+  "platform/ios/MapLibre.docc/GettingStarted.md",
+  "platform/ios/MapLibre.docc/LineOnUserTap.md",
+];
+
+/** source files that contain examples */
+const examplePaths = [
+  "platform/ios/app-swift/Sources/SimpleMapView.swift",
+  "platform/ios/app-swift/Sources/LineTapMapView.swift"
+];
+
+/**
+ * Parses a source file and looks for examples fenced in
+ * ```
+ * // #-example-code(ExampleId)
+ * ...
+ * // #-end-example-code
+ * ```
+ * @param {string} examplePath
+ * @returns Mapping from example id to example, e.g. {"ExampleId": "..."}
+ */
+async function parseExampleCode(examplePath) {
+  const content = await fs.readFile(examplePath, "utf8");
+  const lines = content.split("\n");
+  /** @type {Record<string, string>} */
```

Comment:
Why not write it in typescript instead of this comment?

Reply:
I don't want to complicate our build setup further by requiring TypeScript compilation.

Reply:
Not sure if a lot of people are familiar with using TypeScript with JSDoc, but plain JavaScript remains an option. We're not building a library with JavaScript/TypeScript, just some auxiliary scripts.

Reply:
Since it's a single script you can run it using ts-node without something special, but your call, I added this remark because of the comment here. 
Your call.